pkg/generic/httpthrift_codec.go

```diff
@@ -46,15 +47,41 @@ type HTTPRequest = descriptor.HTTPRequest
 type HTTPResponse = descriptor.HTTPResponse
 
 type httpThriftCodec struct {
-	svcDsc           atomic.Value // *idl
-	provider         DescriptorProvider
-	codec            remote.PayloadCodec
-	binaryWithBase64 bool
+	svcDsc                   atomic.Value // *idl
+	provider                 DescriptorProvider
+	codec                    remote.PayloadCodec
+	binaryWithBase64         bool
+	dyConvOpts               conv.Options
+	dyConvOptsWithThriftBase conv.Options
+	t2jBinaryConv            t2j.BinaryConv
+	dynamicgoEnabled         bool
+	enableDynamicGoHTTPResp  bool
+	fallback                 bool // fallback will be true when dynamicgo is expected to be used but it cannot be
 }
 
-func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec) (*httpThriftCodec, error) {
+func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec, opts *Options) (*httpThriftCodec, error) {
 	svc := <-p.Provide()
-	c := &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false}
+	c := &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false, dynamicgoEnabled: false, fallback: false}
```

Comment:
why do these two have to be taken apart? Can't they be combined?

Reply:
dynamicgoEnabled: false, fallback: false

Reply:
fixed